framework/src/packets/buffer.rs

```diff
@@ -174,7 +174,7 @@ mod tests {
 
                 assert!(alloc(mbuf, 16, 8).is_ok());
                 assert_eq!(24, (*mbuf).data_len());
-                let slice = &(*read_slice(mbuf, 0, 24).unwrap());
+                let slice = &(*read_slice::<u8>(mbuf, 0, 24).unwrap());
```

Comment:
needed the specific casts?

Reply:
yeah I was really confused. it was working fine before, but after I cleaned up the crates, all of a sudden `rustc` freaks out and can't infer types on these calls anymore.

Reply:
gotcha.